suite/tests/common/getretaddr.c

```diff
@@ -113,8 +113,11 @@ GLOBAL_LABEL(FUNCNAME:)
         ldr      x30, [sp], #16
         ret
 # elif defined(RISCV64)
-        /* TODO i#3544: Port tests to RISC-V 64 */
-        ret
+        mv       t0, ra
+        call     next_instr
+    next_instr:
+        mv       a0, ra
```

Comment:
(No action requested)

I'm a little confused by this routine that's called `get_retaddr` but seems to get the retaddr of an internal call instr that points to the instr at `next_instr`. I don't see how that's useful for the caller.

I guess the precise returned value is not really what the caller wants; maybe the goal is only to verify that this sequence of instrs work under DR (as noted by the "make sure dynamorio can handle this non-call" comment).

Reply:
> (No action requested)
> 
> I'm a little confused by this routine that's called `get_retaddr` but seems to get the retaddr of an internal call instr that points to the instr at `next_instr`. I don't see how that's useful for the caller.

So do I. The return value isn't really used in the test.

> I guess the precise returned value is not really what the caller wants; maybe the goal is only to verify that this sequence of instrs work under DR (as noted by the "make sure dynamorio can handle this non-call" comment).